src/model/changes.utils.ts

```diff
@@ -0,0 +1,62 @@
+import { cloneDeepWith } from 'lodash';
+import { EditorState } from 'prosemirror-state';
+import {JSONObject, Manuscript} from './manuscript';
+import {ProsemirrorChange} from "./history/prosemirror-change";
+import { Change } from './history/change';
+import {BatchChange} from "./history/batch-change";
+
+export function manuscriptEntityToJson<T>(object: T): JSONObject {
+  return cloneDeepWith(object, (value) => {
+    if (value instanceof EditorState) {
+      return value.toJSON();
+    }
+  });
+}
+
+export function deserializeChanges(changesJson: JSONObject[]): Array<Change>  {
+  return changesJson.map((changeData) => {
+    switch (changeData.type) {
+      case 'prosemirror':
+        return ProsemirrorChange.fromJSON(changeData);
+      case 'batch':
+        return BatchChange.fromJSON(changeData);
+
+      default:
+        return null;
+    }
+  }).filter(Boolean) as Change[];
+}
+
+export function cloneManuscript(manuscript: Manuscript): Manuscript {
+  const cloneCustomizer = (value: any): EditorState | undefined => (value instanceof EditorState ? value : undefined);
```

Comment:
@ilya-zhuravel I'm not hugely familiar with using the customizer callback with `cloneDeepWith` and it looks like it's doing something different to how I thought it might be working would you mind giving me a run down on how the return `undefined` works? (I assumed it was a simple recursive transform callback but it doesn't look like that's the case as the unit test I've written still has values for none `EditorState` properties).